HW2/P3/P3.txt

```diff
@@ -0,0 +1,43 @@
+--- Multithreading without Instruction-level Parallelism ---
+
+For now we just compare the code efficiency when only the rows parallelized with a prange.
+
+1 thread:
+
+1074.656613 Million Complex FMAs in 4.0765440464 seconds, 263.619526925 million Complex FMAs / second
+
+2 threads:
+
+1074.656613 Million Complex FMAs in 2.06194496155 seconds, 521.185886646 million Complex FMAs / second
+
+4 threads:
+
+1074.656613 Million Complex FMAs in 1.10520887375 seconds, 972.356120662 million Complex FMAs / second
+
+Because this problem is embarassingly parallel, multitheading reduces the run time: 2 threads divides the run time roughly by 2, 4 threads divide the run time roughly by 4.
+
+A strange thing happens with prange when using 1 thread. The code is slower with 1 thread and prange than when using  a simple range without parallelizing.
```

Comment:
Ray mentioned that this could be because when you use prange, even with only one thread, the compiler gets confused and doesn't make certain optimizations because it's not sure that they're thread-safe.